ogr/services/github/service.py

```diff
@@ -47,64 +56,53 @@ def __init__(
         github_app_id: str = None,
         github_app_private_key: str = None,
         github_app_private_key_path: str = None,
+        tokman_instance_url: str = None,
+        github_authentication: GithubAuthentication = None,
         **_,
     ):
         super().__init__()
-        self.token = token
-
-        # Authentication via GitHub app
-        self.github_app_id = github_app_id
-        self._github_app_private_key = github_app_private_key
-        self.github_app_private_key_path = github_app_private_key_path
-
-        self.github = github.Github(login_or_token=self.token)
         self.read_only = read_only
+        self.authentication = github_authentication
+
+        if not self.authentication:
+            self.authentication = self.__get_authentication(
+                token={"token": token} if token else None,
+                github_app={
+                    "id": github_app_id,
+                    "private_key": github_app_private_key,
+                    "private_key_path": github_app_private_key_path,
+                }
+                if github_app_id
+                else None,
+                tokman_instance_url={"instance_url": tokman_instance_url}
+                if tokman_instance_url
+                else None,
```

Comment:
Does it make sense to do this magic? I am only thinking if we can directly create that instance when we have those `if`s here.

If the check for multiple methods is the problem. Define priorities in docstring and skip it...;)